pkg/engines/slurm/slurm.go

```diff
@@ -140,55 +141,52 @@ func GenerateOutput(ctx context.Context, tree *topology.Vertex, params map[strin
 
 func GenerateOutputParams(ctx context.Context, tree *topology.Vertex, params *Params) ([]byte, error) {
 	buf := &bytes.Buffer{}
-	path := params.TopoConfigPath
+	path, plugin := params.TopoConfigPath, params.Plugin
+
+	// set and validate plugin
+	switch plugin {
+	case "":
+		plugin = topology.TopologyTree
```

Comment:
If no plugin is passed, shouldn't we check if block subtree exists? If it does, default should be blockTopology.

Reply:
For Kubernetes, block topology won't provide enough information.
For Kubernetes we always want to provide tree topology, because we are labeling the nodes according to switch hierarchy.